girder/models/model_base.py

```diff
@@ -45,6 +45,44 @@
 _modelSingletons = []
 
 
+def _permissionClauses(user=None, level=None, prefix=''):
+    """
+    Given a user and access level, return a list of clauses that can be used as
+    part of a Mongo find query or aggregate match step.
+
+    :param user: The user to check policies against.
+    :type user: dict or None
+    :param level: The access level.  Explicitly passing None skips doing
+        permissions checks.
+    :type level: AccessType
+    :param prefix: an optional string to prepend to the keys used in the
+        clauses.
+    :type prefix: str
+    :returns: A query dictionary with an '$or' entry which consists of a list
+        of match clauses, any one of which implies validation.
+    """
+    permissionClauses = []
+    if level is None or (user and user['admin']):
+        # Without a level or with an admin user, match everything.
+        permissionClauses.append({})
+    elif level <= AccessType.READ:
+        permissionClauses.append({prefix + 'public': True})
+    elif not user:
+        # If we have no user and asked for higher than read access, make a
+        # query that will fail
+        permissionClauses.append({'__matchnothing': 'nothing'})
```

Comment:
Also since we are `$or`ing these together, doesn't that mean this won't change any behavior?

Reply:
Something has to return no data when there is no user and a above read level.  We can certainly use a different permission clause, but we have to have *some* permission clause or have a special case where sometimes we return a cursor and sometimes we don't.  Instead of this being a special case wherever permission clauses might be needed, this is now in a single location where we generate the permission clause.

Reply:
In this case there will be exactly one clause that gets `$or`ed, which requires that at least one clause applies.  So it will get used.

Reply:
To make that more explicit / readable, can we `return` in this block instead of appending to the list of clauses?

Reply:
Will do.